pkg/rbac/role_binding.go

```diff
@@ -78,3 +78,11 @@ func DeleteRoleBinding(c base.K8sClientWithLogging, ctx context.Context, namespa
 	}
 	return nil
 }
+
+// Validate ensures the role binding is not to cluster-admin
+func (rb *RoleBinding) Validate() error {
+    if rb.RoleRef.Name == "cluster-admin" {
+        return fmt.Errorf("role binding cannot be to cluster-admin")
```

Comment:
As said above, when do we want to use this validation function? There are likely some cases where we do want to bind this role. Especially in a namespace it seems reasonable.